crates/core/src/bcudp/xml_crypto.rs

```diff
@@ -0,0 +1,34 @@
+const XML_KEY: [u32; 8] = [
+    0x1f2d3c4b, 0x5a6c7f8d, 0x38172e4b, 0x8271635a, 0x863f1a2b, 0xa5c6f7d8, 0x8371e1b4, 0x17f2d3a5,
+];
+
+pub(crate) fn decrypt(offset: u32, buf: &[u8]) -> Vec<u8> {
+    let key = XML_KEY
+        .iter()
+        .map(|i| (i + offset).to_le_bytes())
```

Comment:
Can you express the key array as bytes, so you don't need to do little-endian type conversions? You can still group them visually in the source code

Reply:
Again this comes from [here](https://github.com/jdillenkofer/camera_proxy) I tried to get it to do it in just bytes but it kept failing to work right, so I just got it working in the same way it was implemented there thinking to come back when I have time.

Reply:
Perhaps my bitwise operators are not good or something, but how do I do a byte wise addition of two u32s if I break it up into u8s? The `offset` is added to the `XML_KEY` which can make it carry the value to the next byte. I could represent it as u8 bytes then, loop them four at a time while reconstucting the u32 for the addition, but this is visually more complex then just the `to_le_bytes`

Reply:
I could also assume no carry such that the last bit of the byte is zero. But I might need to adjust how the offset is made in other parts of the code

Reply:
Oh I see. You're correct, it needs to be u32 because of the addition!